lint/lint.go

```diff
@@ -28,6 +29,17 @@ type Context struct {
 
 	// TypesInfo carries parsed packages types information.
 	TypesInfo *types.Info
+
+	// Warnings contains warnings from all checkers
+	Warnings []Warning
+
+	mutex sync.Mutex
```

Comment:
There is a convention to place mutex that protects struct fields above those fields with a special comment.
Do something like this:
```go
mutex sync.Mutex // Protects the rest of the fields

// Warnings contains warnings from all checkers
Warnings []Warning
```

You can find many examples in stdlib. Do a grep from `GOROOT/src` like this:
```
grep -nr -C 3 'sync.Mutex'
```

Another trick is to use anonymous struct that embeds mutex and contains named "guarded" fields.
It's overkill for this case as we may change the exact sync patterns and maybe move to checker-local storage again (but this time, no warnings inside checker itself) to avoid any sync at all.